tests/apps/test_data.py

```diff
@@ -328,3 +332,111 @@ def test_data_dim_correct(self, dataset_patched):
+        yield datasets(t=0)
+
+    def test_w_dims(self, dataset):
+        """Test if w has correct shape"""
+        w = dataset.w
+        assert w.shape == (dataset.modes,)
+
+    def test_u_dims(self, dataset):
+        """Test if U has correct shape"""
+        U = dataset.U
+        assert U.shape == (dataset.modes, dataset.modes)
+
+    def test_u_unitary(self, dataset):
+        """Test if U is unitary"""
+        U = dataset.U
+        assert np.allclose(np.eye(U.shape[0]), U.conj().T @ U)
+
+    def test_sample_dims(self, dataset):
+        """Test if sample has correct shape"""
+        assert np.shape(dataset[:]) == (5000, dataset.modes)
+
+    def test_times_water(self, dataset):
+        """Test if available times are correct"""
+        t = dataset.available_times()
+        assert np.allclose(t, np.linspace(0, 270, 28))
+
+    def test_dataset_water(self, datasets):
+        """Test if function raises a ``ValueError`` when an incorrect time is given"""
+        with pytest.raises(ValueError, match="The selected time is not correct"):
+            datasets(t=7)
+
+
+@pytest.mark.parametrize("datasets", PYRROLE_DATASETS_LIST)
+class TestPyrroleDatasets:
+    """Tests for the ``Pyrrole`` class"""
+
+    @pytest.fixture
+    def dataset(self, datasets):
+        """Fixture for loading each of the datasets in ``Pyrrole_DATASETS_LIST``"""
+        yield datasets(t=0)
+
+    def test_ri_dims(self, dataset):
+        """Test if ri has correct shape"""
+        ri = dataset.ri
+        assert ri.shape == (30,)
+
+    def test_rf_dims(self, dataset):
+        """Test if rf has correct shape"""
+        rf = dataset.rf
+        assert rf.shape == (30,)
+
+    def test_wi_dims(self, dataset):
+        """Test if wi has correct shape"""
+        wi = dataset.wi
+        assert wi.shape == (dataset.modes,)
+
+    def test_wf_dims(self, dataset):
+        """Test if wf has correct shape"""
+        wf = dataset.wf
+        assert wf.shape == (dataset.modes,)
+
+    def test_m_dims(self, dataset):
+        """Test if m has correct shape"""
+        m = dataset.m
+        assert m.shape == (30,)
+
+    def test_Li_dims(self, dataset):
+        """Test if Li has correct shape"""
+        Li = dataset.Li
+        assert Li.shape == (30, dataset.modes)
+
+    def test_Lf_dims(self, dataset):
+        """Test if Lf has correct shape"""
+        Lf = dataset.Lf
+        assert Lf.shape == (30, dataset.modes)
+
+    def test_up_dims(self, dataset):
+        """Test if U has correct shape"""
+        U = dataset.U
+        assert U.shape == (dataset.modes, dataset.modes)
+
+    def test_up_unitary(self, dataset):
+        """Test if U is unitary"""
+        U = dataset.U
+        assert np.allclose(np.eye(U.shape[0]), U.conj().T @ U)
+
+    def test_samplep_dims(self, dataset):
+        """Test if sample has correct shape"""
+        assert np.shape(dataset[:]) == (1000, dataset.modes)

```

Comment:
Also here, shouldn't it be `(dataset.n_samples, dataset.modes)`
With 10k samples?

Reply:
I believe the tests are happening for a specific time, namely `t=0`, so the n_samples is 1000